### PR TITLE
rgw: enrich the failed debug log line for sync notification

### PR DESCRIPTION
**Pull request by kchheda3**
RGW: enrich the failure log line for sync notification with bucket, object and owner name.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_notify.cc [hunk 1]
```diff
@@ -990,16 +990,25 @@ int publish_commit(rgw::sal::Object* obj,
         const auto ret = push_endpoint->send_to_completion_async(
 	  dpp->get_cct(), event_entry.event, res.yield);
         if (ret < 0) {
-          ldpp_dout(dpp, 1) << "ERROR: push to endpoint "
-			    << topic.cfg.dest.push_endpoint
-			    << " failed. error: " << ret << dendl;
+          ldpp_dout(dpp, 1)
+              << "ERROR: push to endpoint " << topic.cfg.dest.push_endpoint
+              << " bucket: " << event_entry.event.bucket_name
```
Comment (BBoozmen) [inline src/rgw/driver/rados/rgw_notify.cc:995]: These log lines (lines 995-998 and 1004-1011) have whitespace both at the beginning and at the end - leading to having 2 spaces between words. I think we can get rid of one of them at these log lines.

File: src/rgw/driver/rados/rgw_notify.cc [hunk 2]
```diff
@@ -990,16 +990,25 @@ int publish_commit(rgw::sal::Object* obj,
...
           if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_failed);
           return ret;
         }
         if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_ok);
       } catch (const RGWPubSubEndpoint::configuration_error& e) {
-        ldpp_dout(dpp, 1) << "ERROR: failed to create push endpoint: " 
-            << topic.cfg.dest.push_endpoint << ". error: " << e.what() << dendl;
+        ldpp_dout(dpp, 1) << "ERROR: failed to create push endpoint: "
+                          << topic.cfg.dest.push_endpoint
+                          << topic.cfg.dest.push_endpoint
```
Comment (BBoozmen) [inline src/rgw/driver/rados/rgw_notify.cc:1006]: This line repeats the previous.

File: src/rgw/driver/rados/rgw_notify.cc [hunk 3]
```diff
@@ -990,16 +990,25 @@ int publish_commit(rgw::sal::Object* obj,
...
+                          << " bucket: " << event_entry.event.bucket_name
+                          << " bucket_owner: "
+                          << event_entry.event.bucket_ownerIdentity
+                          << " object_name: " << event_entry.event.object_key
+                          << ". error: " << e.what() << dendl;
```
Comment (BBoozmen) [inline src/rgw/driver/rados/rgw_notify.cc:1011]: extra "." char?

File: src/rgw/driver/rados/rgw_notify.cc (hunk 2 above)
Comment (kchheda3) [reply src/rgw/driver/rados/rgw_notify.cc:1006]: oops.

**Comment by yuvalif** [comment]
thanks for the contribution!
one comment on the headline of the commit/PR.
this is not specific to "sync notifications". any notification that fails will add this info to the debug log.
also, need to be clear that we don't log more info to the notification, but to the debug log :-)
coudl you please change the headline of the commit (and the PR) to: "rgw: enrich the failed notification debug log line"

**Comment by kchheda3** [comment]
> thanks for the contribution! one comment on the headline of the commit/PR. this is not specific to "sync notifications". any notification that fails will add this info to the debug log. also, need to be clear that we don't log more info to the notification, but to the debug log :-) coudl you please change the headline of the commit (and the PR) to: "rgw: enrich the failed notification debug log line"

Thanks for the feedback @yuvalif ,
But i feel this log line is specific only to sync notification ? coz the log line is in the else part of persistent notification check?

**Comment by yuvalif** [comment]
> > thanks for the contribution! one comment on the headline of the commit/PR. this is not specific to "sync notifications". any notification that fails will add this info to the debug log. also, need to be clear that we don't log more info to the notification, but to the debug log :-) coudl you please change the headline of the commit (and the PR) to: "rgw: enrich the failed notification debug log line"
> 
> Thanks for the feedback @yuvalif , But i feel this log line is specific only to sync notification ? coz the log line is in the else part of persistent notification check?

ok, the term "sync" is a bit loaded here... i tought you meant "multisite syncronization notifications".
anyway, you are right, this applies to syncronous (=not persistent) notifications.

**Review by yuvalif — APPROVED** [review]

**Comment by kchheda3** [comment]
> > > thanks for the contribution! one comment on the headline of the commit/PR. this is not specific to "sync notifications". any notification that fails will add this info to the debug log. also, need to be clear that we don't log more info to the notification, but to the debug log :-) coudl you please change the headline of the commit (and the PR) to: "rgw: enrich the failed notification debug log line"
> > 
> > 
> > Thanks for the feedback @yuvalif , But i feel this log line is specific only to sync notification ? coz the log line is in the else part of persistent notification check?
> 
> ok, the term "sync" is a bit loaded here... i tought you meant "multisite syncronization notifications". anyway, you are right, this applies to syncronous (=not persistent) notifications.

Thanks @yuvalif , if you could also help in merging the PR (do not have write permissions)

**Review by thotz — APPROVED** [review]

**Comment by yuvalif** [comment]
teuthology pass: https://github.com/ceph/ceph/pull/52510#issuecomment-1671530058

**Pull request merged**